ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutorTest.java

```diff
@@ -1065,6 +1072,60 @@ public void shouldThrowOnSchemaInferenceMismatchForKey() throws Exception {
     assertThat(e.getMessage(), containsString("ksqlDB Generated: {\"type\":"));
   }
 
+  @Test
+  public void shouldBuildSerdeWithSchemaFullName() throws Exception {
+    // Given:
+    when(srClient.getLatestSchemaMetadata(Mockito.any()))
+        .thenReturn(new SchemaMetadata(1, 1, "\"string\""));
+    when(srClient.getSchemaById(1))
+        .thenReturn(new AvroSchema(RAW_SCHEMA));
+    givenDataSourceWithSchema(
+        TOPIC_NAME,
+        SCHEMA,
+        SerdeFeatures.of(SerdeFeature.UNWRAP_SINGLES),
+        SerdeFeatures.of(),
+        FormatInfo.of(FormatFactory.AVRO.name()),
+        FormatInfo.of(FormatFactory.AVRO.name()),
+        false,
+        false);
+    final ConfiguredStatement<InsertValues> statement = givenInsertValues(
+        allColumnNames(SCHEMA),
+        ImmutableList.of(
+            new StringLiteral("key"),
+            new StringLiteral("str"),
+            new LongLiteral(2L)
+        )
+    );
+
+    // When:
+    executor.execute(statement, mock(SessionProperties.class), engine, serviceContext);
+
+    // Then:
+    verify(keySerdeFactory).create(
+        FormatInfo.of(FormatFactory.AVRO.name(), ImmutableMap.of(
+            AvroProperties.FULL_SCHEMA_NAME,"io.confluent.ksql.avro_schemas.KsqlDataSourceSchema"
```

Comment:
nit: is `io.confluent.ksql.avro_schemas.KsqlDataSourceSchema` the default name? If so, can you change it to something which is not default?

Reply:
Done